crates/languages/src/python.rs

```diff
@@ -791,11 +832,23 @@ impl ToolchainLister for PythonToolchainProvider {
                     });
 
             // Compare project paths against worktree root
-            let proj_ordering = || match (&lhs.project, &rhs.project) {
-                (Some(l), Some(r)) => (r == &wr).cmp(&(l == &wr)),
-                (Some(l), None) if l == &wr => Ordering::Less,
-                (None, Some(r)) if r == &wr => Ordering::Greater,
-                _ => Ordering::Equal,
+            let proj_ordering = || {
+                let lhs_project = lhs
+                    .project
+                    .as_ref()
+                    .or(get_venv_parent_dir(lhs).as_ref())
```

Comment:
`or` is eager (you'll always run `get_venv_parent_dir`, even if `project` is some.

You should use `or_else`.

Reply:
I'm getting this error with `or_else`:
```
cannot return value referencing temporary value
returns a value referencing data owned by the current function (rustc E0515)
```

I'll add a map before `or_else`?